__tests__/db.test.js

```diff
@@ -35,29 +36,87 @@ describe("App", () => {
           .get("/api/topics")
           .expect(200)
           .then(({ body }) => {
-            console.log(body.topics, " <<<< my bodyyy");
             body.topics.forEach((element) => {
               let objKeys = Object.keys(element);
-              expect(objKeys.includes("slug")).toBe(true);
-              expect(objKeys.includes("description")).toBe(true);
+              if (objKeys.length > 0) {
+                expect(objKeys.includes("slug")).toBe(true);
+                expect(objKeys.includes("description")).toBe(true);
+              }
             });
           });
       });
     });
   });
 
   describe("GET /api", () => {
-    test("Responds with a status of 200", () => {
-      return request(app).get("/api").expect(200);
-    });
     test("Response returns an object", () => {
+      //Edit this test description
+      return fs
+        .readFile(`endpoints.json`, "utf-8")
+        .then((data) => {
+          return JSON.parse(data);
+        })
+        .then((data) => {
+          return request(app)
+            .get("/api")
+            .expect(200)
+            .then(({ body }) => {
+              expect(body.endpoints).toEqual(data);
+            });
+        });
+    });
+  });
+  //Challenge 4
+  describe("/api/articles", () => {
+    test('404 - Providing an id for an non-existent article, responds with a "Not found" message.', () => {
       return request(app)
-        .get("/api")
-        .expect(200)
+        .get("/api/articles/19")
+        .expect(404)
         .then(({ body }) => {
-          console.log(body);
+          expect(body.msg).toBe("Not found");
+        });
+    });
 
-          expect(typeof body.endpoints).toBe("object");
+    test("Responds with an article array holding the all the right properties.", () => {
+      return request(app)
+        .get("/api/articles/1")
+        .expect(200)
+        .then(({ body }) => {
+          if (body.articles.length > 0) {
+            body.articles.forEach((element) => {
```

Comment:
you shouldn't need to loop through this since it's just going to be an array with one object inside